new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java

```diff
@@ -394,6 +399,95 @@ public static void listServices(String namespace, String labelSelectors) throws
     }
   }
 
+  /**
+   * List jobs in the given namespace.
+   *
+   * @param namespace in which to list the jobs
+   * @param labelSelectors labels to narrow the list of jobs
+   * @return V1JobList list of {@link V1Job} from Kubernetes cluster
+   * @throws ApiException when list fails
+   */
+  public static V1JobList listJobs(String namespace, String labelSelectors)
+      throws ApiException {
+    V1JobList list;
+    try {
+      BatchV1Api apiInstance = new BatchV1Api(apiClient);
+      list = apiInstance.listNamespacedJob(
+          namespace, // String | name of the namespace.
+          null, // String | pretty print output.
+          null, // Boolean | allowWatchBookmarks requests watch events with type "BOOKMARK".
+          null, // String | The continue option should be set when retrieving more results from the server.
+          null, // String | A selector to restrict the list of returned objects by their fields.
+          labelSelectors, // String | A selector to restrict the list of returned objects by their labels.
+          null, // Integer | limit is a maximum number of responses to return for a list call.
+          "", // String | Shows changes that occur after that particular version of a resource.
+          5, // Integer | Timeout for the list/watch call.
+          Boolean.FALSE // Boolean | Watch for changes to the described resources
+      );
+    } catch (ApiException apex) {
+      logger.warning(apex.getResponseBody());
+      throw apex;
+    }
+    return list;
+  }
+
+  /**
+   * Returns the V1Job object given the following parameters.
+   * @param namespace in which to check for the job existence
+   * @param labelSelectors the labels the job is decorated with, if any
+   * @param jobName name of the job to return
+   * @return V1Job object if found otherwise null
+   * @throws ApiException when there is error in querying the cluster
+   */
+  public static V1Job getJob(String namespace, String labelSelectors, String jobName)
+      throws ApiException {
+    List<V1Job> jobs = listJobs(namespace, labelSelectors).getItems();
+    for (V1Job job : jobs) {
+      if (job.getMetadata().getName().equals(jobName)) {
```

Comment:
null pointer check for job.getMetadata()

Reply:
fixed